### PR TITLE
Support for both JSON and XML error response based on request content…

### PR DESCRIPTION
**Pull request by michaelmajgaard**
#424

### CODE REVIEW

File: source/Energinet.DataHub.PostOffice.Common/Extensions/ExceptionExtensions.cs [hunk 1]
```diff
@@ -37,25 +42,91 @@ public static void Log(this Exception source, ILogger logger)
...
+                    ? await XmlSerializeAsync(error).ConfigureAwait(false)
+                    : JsonSerialize(error);
+
+                return request.CreateResponse(
+                    stream,
+                    returnXml ? MediaTypeNames.Application.Xml : MediaTypeNames.Application.Json,
+                    httpStatusCode);
             }
 
             if (source == null)
```
Comment (FirestarJes) [inline source/Energinet.DataHub.PostOffice.Common/Extensions/ExceptionExtensions.cs:90]: In line 95 the syntax of "is null" is used, select one of them, and use them throughout the file.

File: source/Energinet.DataHub.PostOffice.Common/Extensions/HttpRequestDataExtensions.cs [hunk 2]
```diff
@@ -20,17 +20,20 @@
...
 
 namespace Energinet.DataHub.PostOffice.Common.Extensions
 {
     public static class HttpRequestDataExtensions
     {
-        public static HttpResponseData CreateResponse(this HttpRequestData source, Stream stream, HttpStatusCode statusCode = HttpStatusCode.OK)
+        public static HttpResponseData CreateResponse(this HttpRequestData source, Stream stream, string contentType, HttpStatusCode statusCode = HttpStatusCode.OK)
         {
-            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (source == null)
```
Comment (FirestarJes) [inline source/Energinet.DataHub.PostOffice.Common/Extensions/HttpRequestDataExtensions.cs:31]: The same as in the other file, select one syntax and use that.

**Review by FirestarJes — CHANGES_REQUESTED** [review]

File: source/Energinet.DataHub.PostOffice.Common/Extensions/ExceptionExtensions.cs (hunk 1 above)
Comment (michaelmajgaard) [reply source/Energinet.DataHub.PostOffice.Common/Extensions/ExceptionExtensions.cs:90]: fixd

**Review by FirestarJes — APPROVED** [review]

File: source/Energinet.DataHub.PostOffice.Common/Extensions/HttpRequestDataExtensions.cs (hunk 2 above)
Comment (michaelmajgaard) [reply source/Energinet.DataHub.PostOffice.Common/Extensions/HttpRequestDataExtensions.cs:31]: fixd

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**